amp-pwa-reader/src/js/Article.js

```diff
@@ -138,15 +138,16 @@ class Article {
       return Promise.resolve();
     }
 
-    let offset = (innerWidth * this.card.imageData.ratio) / 2;
-    this.container.style.transform = 'translateY(' + scrollY + 'px)';
-    this.container.style.opacity = 1;
-
     return new Promise(resolve => {
-      this.container.animate([
-        { opacity: 0, transform: 'translateY(' + (offset + scrollY) + 'px)' },
-        { opacity: 1, transform: 'translateY(' + scrollY + 'px)' }
-      ], { duration: this.cssVariables.animationSpeedIn, easing: 'cubic-bezier(0.23, 1, 0.32, 1)' }).onfinish = resolve;
+
+      this._transitionEndCallback = () => {
+        this.container.removeEventListener('transitionend', this._transitionEndCallback);
+        resolve();
+      };
+
+      this.container.addEventListener('transitionend', this._transitionEndCallback, false);
```

Comment:
One constant annoyance: if it so happens that destination == target, transitionend never fires :( Not sure how possible it is here. Typically I'd employ [double tap rule](https://www.youtube.com/watch?v=JmA2WYyw-_A), e.g. via timer.